### PR TITLE
Added TypeScript typings

### PR DESCRIPTION
**Pull request by ItalyPaleAle**
### Changes

This PR adds TypeScript definitions for the library. This was auto-generated from JSDoc, but with some manual fixes applied.

### Testing

N/A

### Checklist

- [X] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)
- [X] I have read the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md)
- [X] All existing and new tests complete without errors

### CODE REVIEW

File: types/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+   * It will validate:
+   * - signature according to the algorithm configured in the verifier.
+   * - if nonce is present and matches the one provided
+   * - if `iss` and `aud` claims matches the configured issuer and audience
+   * - if token is not expired and valid (if the `nbf` claim is in the past)
+   * @param token id_token to verify
+   * @param requestedNonce nonce value that should match the one in the id_token claims
+   * @param cb callback used to notify the results of the validation
+   */
+  verify(token: string, requestedNonce: any, cb: verifyCallback): any;
```
Comment (stevehobbsdev) [inline types/index.d.ts:121]: I think we can type `requestedNonce` as `string` here.

File: types/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+  verify(token: string, cb: verifyCallback): any;
+
+  getRsaVerifier(iss: any, kid: any, cb: any): void;
```
Comment (stevehobbsdev) [inline types/index.d.ts:124]: Let's remove this, it's not part of the public documented API.

**Review by stevehobbsdev — CHANGES_REQUESTED** [review]
Looks good, thanks for the contribution! Just a couple of changes needed.

**Comment by ItalyPaleAle** [comment]
@stevehobbsdev Sorry for the delay. Updates as requested

**Comment by fnagel** [comment]
@stevehobbsdev Any chance to get this merged soon? :-)

**Comment by stevehobbsdev** [comment]
@fnagel Yes - let me put out a release this week 👍🏻

**Comment by fnagel** [comment]
@stevehobbsdev Wonderful :-) Thanks a lot!

**Review by stevehobbsdev — APPROVED** [review]

**Pull request merged**